src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -375,8 +375,7 @@ private SortedSet<Violation> processFile(File file) throws CheckstyleException {
      * @return {@code true} if the file is accepted.
      */
     private boolean acceptFileStarted(String fileName) {
-        final String stripped = CommonUtil.relativizePath(basedir, fileName);
-        return beforeExecutionFileFilters.accept(stripped);
+        return beforeExecutionFileFilters.accept(fileName);
```

Comment:
we have only this filter for beforeExecution 
https://checkstyle.sourceforge.io/filefilters/beforeexecutionexclusionfilefilter.html#BeforeExecutionExclusionFileFilter

if we can not reproduce any problem with it, we can create test Filter that just not working if path is full not stripped a bit.

Reply:
example of test with custom module/check
https://github.com/checkstyle/checkstyle/blob/c362c2234de44eee7692c498d9b3ec95a593aac4/src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java#L1415-L1421

Reply:
> we can create test Filter that just not working if path is full not stripped a bit.

Does this mean we can just use a custom test to check if the file is indeed stripped or null over this specific case ?

Reply:
yes, but try to use existing BeforeExecutionExclusionFileFilter to avoid custom hack modules (filter/check).

Reply:
When I let the filter accept the fileName as it is **without being relativized/stripped** it works just as it would have worked if it got stripped so testing over it being not stripped would never really work 

Almost all the tests dont really have the file stripped it's taken as it is because ``baseDir`` is usually null **so relativizing the file /full/path/to/file would just remain as it is when baseDir is null** in the Checker test cases but even if we make a test with custom ``baseDir`` property it wouldnt differ because the filter accepts the full URIs and process them even if they are not stripped

Reply:
>Though I cant really get the point of why not just use the filter to accept the file name as it is (with or without stripping) ?

I don't remember, but changing it might break existing filters setup, so users will be angry.
Filters usually configured by what user see in output, if output is stripped user setup config the way he see it.

>The filter tests do pass even when not stripping the files thats why its confusing to me and cant really understand what exactly to test over ?

If tests pass, it just means that tests is not good enough to detect it. Whole idea of mutational testing as it most deep dive in meaning of each line of code, not just fact that line was executed (jacoco).

I wish we had pitest before we created such filters, so we would not be able make code that is not very clear. But we in real life, so we need to catch up. We should not remove this code and just need to make test that will explain in what use case it becomes to matter.

Reply:
@MohanadKh03, do you have any ideas how to cover this code? Or do you need help?

Reply:
I honestly am not sure how to approach this right now
Could you please let me know if there is a specific test suite I could go through ? BeforeExecutionFilter or Checker tests ?

Reply:
take this as base for new test - BeforeExecutionExclusionFileFilterTest.testFileExclusion

take a look at CheckerTest.testSetSeverity , attention to `verifyWithInlineXmlConfig` and fact that xml config is located in Input file.

please make your own Input with xml config that has BeforeExecutionExclusionFileFilter as modules under Checker and `baseDir` is defined as property.

https://checkstyle.org/filefilters/beforeexecutionexclusionfilefilter.html#BeforeExecutionExclusionFileFilter
https://checkstyle.org/config.html#Checker_Examples

Violation message will not have full filename if `CommonUtil.relativizePath` is used.
So we can validate that `baseDir` is applied to violation and name is short, not absolute path.


We might need to make method that returns violation from Checker to get access to full name of file in violation. ChekckerTest has a lot of test method as example.

Reply:
Sounds great thanks for your help now I get the xml input's idea 
But I do see something that is not clear yet in the ``testFileExclusion`` it says this is a filtered violation 
https://github.com/checkstyle/checkstyle/blob/9b9d3f945c4e5e098a96f5d9f9cb7def46409234/src/test/resources/com/puppycrawl/tools/checkstyle/filefilters/beforeexecutionexclusionfilefilter/InputBeforeExecutionExclusionFileFilter.java#L15-L18 
even though here it is supposed to be an unfiltered one ?
https://github.com/checkstyle/checkstyle/blob/9b9d3f945c4e5e098a96f5d9f9cb7def46409234/src/test/java/com/puppycrawl/tools/checkstyle/filefilters/BeforeExecutionExclusionFileFilterTest.java#L77-L79

As for the tests I think what needs to be done now is to add a proper match for a file that contains a ``baseDir`` and should be excluded when not relativized in this specific case to avoid the mutation am I right ?

Reply:
>As for the tests I think what needs to be done now is to add a proper match for a file that contains a baseDir and should be excluded when not relativized in this specific case to avoid the mutation am I right ?

Right.

>even though here it is supposed to be an unfiltered one ?

Please send separate PR to remove such misleading comment. :) .